functionary/prompt_template/llama3_prompt_plate_v3.py

```diff
@@ -0,0 +1,317 @@
+from functionary.prompt_template.base_template import PromptTemplate, PYTHON_RUN_SYS_MSG
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+from functionary.prompt_template import prompt_utils
+from functionary.schema import generate_schema_from_functions
+from functionary.openai_types import Function, Tool
+
+SYSTEM_CONTENT = """You are capable of executing available function(s) if required.
+Only execute function(s) when absolutely necessary.
+Ask for the required input to:recipient==all
+Use JSON for function arguments.
+Respond in this format:
+>>>${recipient}
+${content}
+Available functions:
+"""
+
+
+class Llama3TemplateV3(PromptTemplate):
+    version = "v3.llama3"
+    function_separator = ">>>"
+    start_header = "<|start_header_id|>"
+    end_header = "<|end_header_id|>"
+    eos_token = "<|eot_id|>"
+
+    def get_additional_tokens(self) -> List[str]:
+        return []
+
+    def get_assistant_prefixes(self) -> List[str]:
+        return [
+            f"{self.start_header}assistant{self.end_header}\n\n{self.function_separator}"
+        ]
+
+    def get_stop_tokens_for_generation(self) -> List[str]:
+        return [self.eos_token, "<|end_of_text|>"]
+
+    # ["<|eot_id|>", "<|end_of_text|>"]
+
+    def get_force_function_call_prefix(self, function_name: str):
+        return f"{function_name}\n"
+
+    def pre_process_messages_before_inference(self, messages: List[Dict]) -> List[Dict]:
+        """Order the tool results by the order of tool call ids
+
+        Args:
+            messages (List[Dict]): List of messages
+
+        Returns:
+            List[Dict]: List of messages
+        """
+        return prompt_utils.reorder_tool_messages_by_tool_call_ids(messages)
+
+    def convert_message_to_prompt(self, message: Dict) -> str:
```

Comment:
This is highly similar to `llama3_prompt_template.py`. Should we consider implementing a double inheritance structure?

base_template -> llama3_base_template -> llama3_prompt_template + llama3_prompt_template_v3

Reply:
Maybe we can keep this to a future PR

Reply:
I think it would be quite confused with multiple-inheritance, we can move re-usable functions to prompt_utils.py in the future